timer

```diff
@@ -0,0 +1,171 @@
+        chmod a+x install.sh #this might be necessary in your case but wasnt in mine.
+        ./$nameOfInstallFile "update" || exit 1
+        cd ..
+        rm -r -f $repositoryName || { echo "Permissions Error: update succesfull but cannot delete temp files located at ~/$repositoryName delete this directory with sudo"; exit 1; }
+      else
+        exit 1
+      fi
+    else
+      echo "$repositoryName is already the latest version"
+    fi
+  fi
+}
+
+checkInternet()
+{
+  httpGet github.com > /dev/null 2>&1 || { echo "Error: no active internet connection" >&2; return 1; } # query github with a get request
+}
+
+usage()
+{
+  cat <<EOF
+timer
+Description: Enter a number of minutes and get a voice prompt when the minutes expire
+Usage: timer
+Examples:
+timer
+EOF
+}
+
+while getopts "uvh" opt; do
+  case "$opt" in
+    \?) echo "Invalid option: -$OPTARG" >&2
+        exit 1
+        ;;
+    h)  usage
+        exit 0
+        ;;
+    v)  echo "Version $currentVersion"
+        exit 0
+        ;;
+    u)  getConfiguredClient || exit 1
+        checkInternet || exit 1
+        update
+        exit 0
+        ;;
+    :)  echo "Option -$OPTARG requires an argument." >&2
+        exit 1
+        ;;
+  esac
+done
+
+# special set of first arguments that have a specific behavior across tools
+if [[ $# == "0" ]]; then
+  usage ## if calling the tool with no flags and args chances are you want to return usage
+  exit 0
+elif [[ $# == "1" ]]; then
+  if [[ $1 == "update" ]]; then
+    getConfiguredClient || exit 1
+    checkInternet || exit 1
+    update || exit 1
+    exit 0
+elif [[ $1 == "help" ]]; then
+    usage
+    exit 0
+  fi
+fi
+
+## The rest of the conditions and code would go here
+get_time () {
+    read -p "How long? Enter minutes: " the_time
+}
+
+check_for_spd-say () {
+if [[ $(which spd-say) == "" ]]
+then
+echo "You need to install spd-say to use this tool. This tool uses speech to alert you."
+echo "Try the command `sudo apt install spd-say`."
+exit 1
+fi
+}
+
+check_for_spd-say
+get_time
+#don't enter a non-number or a negative number
+while ! [ "$the_time" -ge 0 ] 2> /dev/null # must use single brackets
+    do
+    if [ "$the_time" -ge 0 ] 2> /dev/null # must use single brackets
+    then
+       true  # perhaps not needed...
+    else
+       echo "There is something wrong with your input. Enter a positive number."
+       get_time
+    fi
+done
+
+the_time=$((the_time*60))
+
+# a custom message is useful when running more than one timer: "Check the BBQ" "The laundry is done" "Jane should be home soon" etc.
+read -p "Custom message? The default message is 'Time's up!' Press enter for the default or enter a custom message if you want: " the_message
+
+if [[ $the_message == "" ]]
+then
+    the_message="Time's up."
+fi
+(sleep $the_time && spd-say -r -40 -p -80 -t female3 "$the_message")&   # see the spd-say man page for options
```

Comment:
Obviously this whole logic with what is being said will have to change

Reply:
I'm not really sure what you mean here, Alex.

The logic of having it run in the background is good, I think. That way you can schedule multiple reminders on a single command line. You can close the terminal, the process still runs, and the message plays on time. I have tried it. 

If you mean the use of spd-say, which I agree is not a perfect method, then consider how it will work if it only prints the reminder text on the terminal and you are using the terminal for another purpose at the time, you are not watching or you have closed it. And some terminals beep and some don't, it can be a configuration issue.

I appreciate your thoughts. Thank you.

Reply:
Yeah I meant just because if we try to get rid of the spd-say dependency this will have to change.